hatice-ecevit-2016400138/assignment4/countIslands.java

```diff
@@ -0,0 +1,32 @@
+
+public class countIslands {
+    /**
+     *
+     * @param map is the map of tiles given.
+     * @param row number of rows.
+     * @param column number of columns.
+     * @return -1 if the map is null, number of islands otherwise.
+     */
+    public static int findNumOfIslands (boolean[][] map, int row, int column) {
+        if (map == null)
+            return -1;
+        int numOfIslands = 0;
+        for (int i = 0; i < row; i++) {
+            for (int j = 0; j < column; j++) {
+                if (map[i][j]) {
+                    if (i > 0 && j > 0 && map[i-1][j] && map[i][j-1] && !map[i-1][j-1])

```

Comment:
This code is quite hard to follow the logic of for a reviewer. Imagine you were reading it for the first time. Can you do anything to make it more understandable? Options:

- comments
- helper methods (eg getNeighbours(), hasNeighbourWest() , ...)